File: problems/001-two-sum/solution.py
def two_sum(self, nums: list[int], target: int) -> tuple[int, int]:
    seen = {}
    for i, n in enumerate(nums):
        diff = target - n
        if diff in seen:
            return seen[diff], i
        seen[n] = i

    raise ValueError("no solution found")

File: problems/001-two-sum/solution.c
#include <stdio.h>


int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    *returnSize = 2;

    for (int i = 0; i < numsSize; i++)
        for (int j = 0; j < numsSize; j++) {
            if (nums[i] + nums[j] == target && i != j) {
                int* result = malloc(sizeof(int) * 2);
                result[0] = i;
                result[1] = j;
                return result;
            }
        }

    return NULL;
}